package/rpm/build.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+# get version slug
+version_hash=$(git describe --match "v*" | sed 's/^v//' | sed 's/-/./g')
+echo "** repository hash: ${version_hash} ..."
+
+# archive reository
+(cd ../.. && git archive --format=tar.gz --prefix=corrade-${version_hash}/ -o ~/rpmbuild/SOURCES/corrade-${version_hash}.tar.gz HEAD)
+echo "** created archive: ~/rpmbuild/SOURCES/corrade-${version_hash}.tar.gz"
+sleep 2
+
+# replace spec version
+spec_file="corrade.spec"
+cp ${spec_file} ${spec_file}.build
```

Comment:
I'm actually fine with the `corrade.spec` file getting overwritten by `sed` in-place -- it's all tracked by Git anyway, so there's no chance of accidental information loss.

ArchLinux does the same to the `PKGBUILD` files when building them and it has a nice side effect of knowing for what revision was the package script last updated. (And yes, just commit the `*.spec` file with that updated version hash.)